fhir-resource-wih/src/main/java/io/elimu/a2d2/fhirresourcewih/OperationOutcomeWorkItemHandler.java

```diff
@@ -0,0 +1,205 @@
+package io.elimu.a2d2.fhirresourcewih;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemHandler;
+import org.kie.api.runtime.process.WorkItemManager;
+
+import ca.uhn.fhir.context.FhirContext;
+import io.elimu.a2d2.exception.WorkItemHandlerException;
+import io.elimu.a2d2.genericmodel.ServiceResponse;
+
+public class OperationOutcomeWorkItemHandler implements WorkItemHandler {
+
+	public static final String ISSUE_PREFIX = "issue_";
+	
+	@Override
+	public void executeWorkItem(WorkItem workItem, WorkItemManager manager) {
+		String version = (String) workItem.getParameter("fhirVersion");
+		Object status = workItem.getParameter("response_status");
+		Set<String> issueIds = workItem.getParameters().keySet().stream().
+				filter(e -> e.startsWith(ISSUE_PREFIX)).
+				map(s -> s.replace(ISSUE_PREFIX, "")).
+				map(s -> s.substring(0, s.indexOf("_"))).
+				distinct().
+				collect(Collectors.toSet());
+		Object outcomeObj = null;
+		String outcomeJson = null;
+		if (version == null || "FHIR4".equalsIgnoreCase(version) || "R4".equalsIgnoreCase(version)) {
+			org.hl7.fhir.r4.model.OperationOutcome outcome = new org.hl7.fhir.r4.model.OperationOutcome();
+			for (String issueId : issueIds) {
+				Map<String, Object> subset = extractIssueDetails(workItem.getParameters(), issueId);
+				org.hl7.fhir.r4.model.OperationOutcome.OperationOutcomeIssueComponent issue = outcome.addIssue();
+				if (subset.containsKey("code")) {
+					String code = (String) subset.get("code");
+					issue.setCode(org.hl7.fhir.r4.model.OperationOutcome.IssueType.fromCode(code));
+				}
+				if (subset.containsKey("details") || subset.containsKey("detailsCode") || subset.containsKey("detailsSystem")) {
+					String details = (String) subset.get("details");
+					String code = (String) subset.get("detailsCode");
+					String system = (String) subset.get("detailsSystem");
+					org.hl7.fhir.r4.model.CodeableConcept cc = new org.hl7.fhir.r4.model.CodeableConcept();
+					if (code != null || system != null) {
+						org.hl7.fhir.r4.model.Coding coding = cc.addCoding();
+						if (code != null) {
+							coding.setCode(code);
+						}
+						if (system != null) {
+							coding.setSystem(system);
+						}
+					}
+					if (details != null) {
+						cc.setText(details);
+					}
+					issue.setDetails(cc);
+				}
+				if (subset.containsKey("diagnostics")) {
+					issue.setDiagnostics((String) subset.get("diagnostics"));
+				}
+				if (subset.containsKey("location")) {
```

Comment:
location is deprecated in R4. I wonder if we should get rid of it now.

Reply:
@boxwala as long as it is a usable property, I suggest we keep it, and suggest not using it in the documentation in the wiki page